gbe/views/act_tech_wizard_view.py

```diff
@@ -0,0 +1,277 @@
+            if rehearsal_form.cleaned_data['booking_id']:
+                bookable.booking_id = \
+                    rehearsal_form.cleaned_data['booking_id']
+            response = set_act(
+                occurrence_id=rehearsal_form.cleaned_data['rehearsal'],
+                act=bookable)
+            # errors are internal, and not OK to show regular user
+            if response.errors:
+                error = True
+            else:
+                show_general_status(request, response, self.__class__.__name__)
+
+            if response.occurrence:
+                bookings += [response.occurrence]
+                self.rehearsals[int(rehearsal_form.prefix)] = ScheduleItem(
+                    event=response.occurrence,
+                    booking_id=response.booking_id)
+        return error, bookings, forms, request
+
+    def make_context(self, basic_form, rehearsal_forms=None):
+        basic_instructions = UserMessage.objects.get_or_create(
+                view=self.__class__.__name__,
+                code="BASIC_INSTRUCTIONS",
+                defaults={
+                    'summary': "Basic Instructions",
+                    'description': default_basic_acttech_instruct})
+        rehearsal_instruct = UserMessage.objects.get_or_create(
+                view=self.__class__.__name__,
+                code="REHEARSAL_INSTRUCTIONS",
+                defaults={
+                    'summary': "Rehearsal Instructions",
+                    'description': default_rehearsal_acttech_instruct})
+        if not rehearsal_forms:
+            rehearsal_forms = self.set_rehearsal_forms()
+        if basic_form:
+            self.first_title = "Change Rehearsal"
+        context = {'act': self.act,
+                   'shows': self.shows,
+                   'rehearsals': self.rehearsals,
+                   'rehearsal_forms': rehearsal_forms,
+                   'second_form': basic_form,
+                   'page_title': self.page_title,
+                   'first_title': self.first_title,
+                   'second_title': self.second_title,
+                   'basic_instructions': basic_instructions[0].description,
+                   'rehearsal_instructions': rehearsal_instruct[0].description}
+        return context
+
+    def groundwork(self, request, args, kwargs):
+        self.shows = []
+        self.rehearsals = {}
+        profile = validate_profile(request, require=False)
+        if not profile:
+            return HttpResponseRedirect(reverse('profile_update',
+                                        urlconf='gbe.urls'))
+        act_id = kwargs.get("act_id")
+        self.act = get_object_or_404(Act, id=act_id)
+        if self.act.performer.contact != profile:
+            validate_perms(request, self.permissions)
+        response = get_schedule(labels=[self.act.b_conference.conference_slug],
+                                act=self.act)
+
+        for item in response.schedule_items:
```

Comment:
What happens if there are no shows?  ie, act not accepted for anything.  
Answer - if you get the url, it shows you the tech form, but no rehearsal options.
TODO - correct this, give a 404 error instead (testing in progress)